code/controllers/subsystems/air.dm

```diff
@@ -423,15 +423,15 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
 		return edge
 
 /datum/controller/subsystem/air/proc/has_same_air(turf/A, turf/B)
-	if(A.oxygen != B.oxygen) 
+	if(A?.oxygen != B?.oxygen) 
```

Comment:
Bad place for the `?.` operator. It's possible that _both_ are `null`, this results in everything passing and the proc STILL returning `1`. Simply add this as a front guard:
```dm
if (!A || !B)
    return 0
```